### ISSUE TITLE
[feature] provide more details in error message when an option is missing

### ISSUE THREAD

**Issue by Marcin-Radecki** [conan-io/conan#6373]
Hi, I have proposal for small enhancement when with regards to verbosity of error message when an option is missing. Let's assume we have option `bar` defined in the conanfile of package `foo`, and no `default_options` defined. In the conanfile, an expression `self.options.bar` is used. When I try to create package without specifying `-o bar=baz`, the command fails as expected:

```
conan create . test/test
...
ERROR: foo/1.0@test/test: 'bar' value not defined
```

Though the error is rather cryptic in my opinion. In particular, it does not say that an _option_ is missing. Also, it does not say what are the possible `bar` values.  In my option, it we'd be nice if conan by default specify those pieces of information in such use case. 

Above behavior can be implemented in the same conanfile using below snippet:

```
def configure(self):
    if ('bar', None) in self.options.items():
        raise Exception("Missing option bar. Possible values: [baz]")
```

Conan version: 1.20.2
- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by jgsogo** [comment]
Hi! Yes, this should be really easy to improve. Thanks for the tip!

**Comment by memsharded** [comment]
Message is fixed in 2.0, saying clearly it is an "option" without a value, closing as solved
